controllers/ProposalController.php

```diff
@@ -676,4 +747,109 @@ private function saveEditedComment(Comment $comment, $newCommentContent)
             throw new CannotSaveException($comment);
         }
     }
+
+    /**
+     * Display proposals for a Publisher
+     * or single proposal
+     *
+     * @param null $id
+     * @return string
+     */
+    public function actionManageProposals($id = null)
+    {
+        if (!is_null($id)) {
+            $selectedProposal = $this->checkIfProposalExists($id);
+            $displayItems = $this->buildOneProposalDisplayItems($selectedProposal);
+
+            return $this->render('proposal', $displayItems);
+        }
+
+        $approvedProposalsQuery = $this->buildApprovedProposalsQuery();
+        $approvedProposals = $this->buildApprovedProposalsActiveDataProvider($approvedProposalsQuery);
+        $notApprovedProposals = $this->getNotApprovedProposals($approvedProposalsQuery);
+
+        return $this->render('manage-proposals', [
+           'approvedProposals' => $approvedProposals,
+           'notApprovedProposals' => $notApprovedProposals
+        ]);
+    }
+
+    private function buildApprovedProposalsQuery()
+    {
+        return \app\models\Proposal::find()
+            ->select('proposal.*, (SELECT COUNT(*) FROM review WHERE review.proposal_id = proposal.id) as count_reviews')
+            ->where('1 = CASE 
+                WHEN 
+                    ((((SELECT COUNT(*) FROM review 
+                        WHERE review.status = \'approved\' AND review.proposal_id = proposal.id)
+                        /
+                        (SELECT COUNT(*) FROM review
+                        WHERE review.proposal_id = proposal.id))
+                    *100)  
+                    >= ' . yii\helpers\Html::encode(ProposalApprovementSetting::getApprovementPercent()). ') 
```

Comment:
Why do you Html::encode this value ? It's not displayed in a HTML page

Reply:
Later, admins will have the possibility to edit this. As this time we don't know if it's gonna be text input or slider (or else), so I choose to secure this field in case of text input.